### ISSUE TITLE
Freeze when dragging tabs (confirmed with 1.32.1)

### ISSUE THREAD

**Issue by PerBothner** [atom/tabs#544]
### Prerequisites

* [X ] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

Atom completely freezes when dragging tab.

### Steps to Reproduce

1. Start atom with no options (or with `--safe`). (The problem goes away with `--disable-gpu`.)
2. Drag a pane.

**Expected behavior:**
Moved pane.

**Actual behavior:**
Atom hangs.  Sometimes it steals the window cursor, making it hard to kill.

**Reproduces how often:**
100%

### Versions
Atom version is 1.32.1, amd64, from atom.io.
OS is Fedora 29, with Gnome and Wayland.

### Additional Information
This appears to be Issue #541, confirmed with latest official release.

**Comment by duvallj** [comment]
Also getting the exact same thing on ArchLinux, x86_64, X.org 1.20.3, with `atom --version`

```
Atom    : 1.32.1
Electron: 2.0.12
Chrome  : 61.0.3163.100
Node    : 8.9.3
```

Interestingly, allowing atom to use a dedicated NVIDIA GPU (with bumblebee + `optirun`) instead of the integrated Intel GPU also causes the problem to go away.

**Comment by PerBothner** [comment]
I have used older versions of Atom where this did not happen.  Can't say when it started - I don't use Atom on a daily basis.  I'm guessing happened in the last new months.

**Comment by rsese** [comment]
It looks like https://github.com/electron/electron/issues/12820 mentions that it's specific to Electron 2 and above and Atom was updated to Electron 2.0 in 1.28.0.  If it's not too much trouble, can you confirm if things work ok in Atom 1.27.0?

**Comment by PerBothner** [comment]
Atom 1.27.2 seems to be fine.

**Comment by rsese** [comment]
Thanks @PerBothner - that's the last release before we moved to Electron 2.0 so it seems like this could be at least related if not the same issue as https://github.com/electron/electron/issues/12820?  I'd assume that you'll see the freeze if you try 1.28.0.

**Comment by PerBothner** [comment]
This is apparently fixed in Electron 3.0.14 https://github.com/electron/electron/issues/12820#issuecomment-452702533

**Comment by rsese** [comment]
:+1:  I noted this issue in https://github.com/atom/atom/pull/18916, the pull request to update to Electron 3.
